File: 1.5-ArrayBasics/ArrayBasics_in_CPP/6-DuplicateInArray.cpp
#include<bits/stdc++.h>
using namespace std;

int findDuplicate(int arr[], int n){
    
    int ans = 0;

    for(int i=0; i<n; i++){
        ans = ans ^ arr[i];
    }

    for(int i=1; i<=n-1; i++){
        ans = ans ^ i;
    }

    return ans;
}

int main() {

    int arr[7] = {6,3,1,5,4,3,2};

    cout<<findDuplicate(arr, 7);

    return 0;
}

File: 1.5-ArrayBasics/ArrayBasics_in_Python/6-DuplicateInArray.py
def findDuplicate(arr):
    
    ans = 0

    for i in range(len(arr)):
        ans = ans ^ arr[i]

    for i in range(1,len(arr)):
        ans = ans ^ i

    return ans 


arr = [6,3,1,5,4,3,2]

print(findDuplicate(arr))